src/formElementPlugins/fileUpload/src/ApplicationView.tsx

```diff
@@ -176,77 +201,20 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
-                  <Grid.Row style={{ boxShadow: 'none', marginLeft: 150 }} verticalAlign="bottom">
-                    <Icon
-                      link
-                      name="delete"
-                      circular
-                      fitted
-                      color="grey"
-                      onClick={() => handleDelete(key)}
-                    />
-                  </Grid.Row>
-                  {error && (
-                    <>
-                      <Grid.Row
-                        centered
-                        style={{ boxShadow: 'none', height: 120, padding: 10 }}
-                        verticalAlign="middle"
-                      >
-                        <Message negative compact>
-                          <p>{errorMessage}</p>
-                        </Message>
-                      </Grid.Row>
-                      <Grid.Row centered style={{ boxShadow: 'none' }}>
-                        <p style={{ wordBreak: 'break-word' }}>{filename}</p>
-                      </Grid.Row>
-                    </>
-                  )}
-                  {loading && (
-                    <>
-                      <Grid.Row
-                        centered
-                        style={{ boxShadow: 'none', height: 120 }}
-                        verticalAlign="middle"
-                      >
-                        <Loader active size="medium">
-                          Uploading
-                        </Loader>
-                      </Grid.Row>
-                      <Grid.Row centered style={{ boxShadow: 'none' }}>
-                        <p style={{ wordBreak: 'break-word' }}>{filename}</p>
-                      </Grid.Row>
-                    </>
-                  )}
-                  {fileData && (
-                    <>
-                      <Grid.Row centered style={{ boxShadow: 'none' }} verticalAlign="top">
-                        <a href={downloadUrl + fileData.fileUrl} target="_blank">
-                          <Image
-                            src={downloadUrl + fileData.thumbnailUrl}
-                            style={{ maxHeight: prefs.applicationViewThumbnailHeight }}
-                          />
-                        </a>
-                      </Grid.Row>
-                      <Grid.Row centered style={{ boxShadow: 'none' }}>
-                        <p style={{ wordBreak: 'break-word' }}>
-                          <a href={downloadUrl + fileData.fileUrl} target="_blank">
-                            {filename}
-                          </a>
-                        </p>
-                      </Grid.Row>
-                    </>
-                  )}
-                </Grid>
-              </List.Item>
+        <List className="file-list" horizontal={!showDescription} verticalAlign="top">
+          {uploadedFiles.map((file) =>
+            showDescription ? (
+              <FileDisplayWithDescription
+                key={file.key}
```

Comment:
noticed that key is missing from normal FileDisplay, and that both just get the key from `file`. Would suggest to remove `key` from `FileDisplayWithDescription` and get `key` from file and make sure List.Item has key (for react)

Reply:
Doesn't the immediate child of in a component array need the key (for React)? So inside the `<List>` component, the immediate children are the `<FileDisplayWithDescription>` and `<FileDisplay>` components, so they must have a key for React. The `<List.Item>` components within each of those don't need a key as there's only one of them for each "FileDisplay" component. (The "FileDisplay" components are really just a wrapper for "List.Item"). Is that right? 

Anyway, have added key to "FileDisplay": efd92223

Reply:
ok, i wasn't sure about the key being on the immediate child. I just saw it was different between two similar components